### PR TITLE
Close #1734 create a page for the guest card focusing on bib card background and color

### LINKED ISSUE #1734: Create a page for the guest card focusing on BibCard background and color
- [x] Need to be create the table cm_guest_cards that have feilds 'background', 'color' of BibCard and EventCard
- [x] Create a new design page for the GuestCard on the event page that allows the admin to upload a background image and select the color for the BibCard and EventCard.
Screenshot from 2024-08-08 14-25-22

- [ ] Next, connect the color field in GuestCard to Variant so that it can fetch all data from GuestCard.
Screenshot from 2024-08-08 14-55-40

### PR DESCRIPTION
**Pull request by Sreyleak-Deth**
Screenshot from 2024-08-11 00-45-39
Screenshot from 2024-08-10 15-25-27
Screenshot from 2024-08-11 00-35-13
Screenshot from 2024-08-13 16-35-08
Screenshot from 2024-08-13 16-34-59

### CODE REVIEW

File: app/controllers/spree/admin/guest_card_classes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+module Spree
+  module Admin
+    class GuestCardClassesController < Spree::Admin::ResourceController
+      before_action :load_taxonomy_taxon
+
+      def index
+        @guest_card_classes = @taxon.guest_card_classes
+      end
+
```
Comment (channainfo) [inline app/controllers/spree/admin/guest_card_classes_controller.rb:9]: annotate custom action with comment:

\# DELETE /remove_background_image

File: app/models/spree_cm_commissioner/guest_card_classes/bib_card_class.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+module SpreeCmCommissioner
+  module GuestCardClasses
+    class BibCardClass < SpreeCmCommissioner::GuestCardClass
+      belongs_to :taxon, class_name: 'Spree::Taxon', optional: true
+      preference :background_color, :string
+
+      has_one_attached :background_image do |attachable|
+        attachable.variant :mini, resize_to_limit: [437, 622]
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/guest_card_classes/bib_card_class.rb:8]: The same here.

**Review by channainfo — CHANGES_REQUESTED** [review]

**Comment by Sreyleak-Deth** [comment]
Noted bong, I will correct it

File: app/models/spree_cm_commissioner/bib_card_background_image.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module SpreeCmCommissioner
+  class BibCardBackgroundImage < SpreeCmCommissioner::Asset
+    def asset_styles
+      {
+        mini: '400x600>',
+        small: '800x1200>',
```
Comment (channainfo) [inline app/models/spree_cm_commissioner/bib_card_background_image.rb:6]: let's do only two variants only.
400x600
800x1200

**Review by channainfo — CHANGES_REQUESTED** [review]

**Pull request merged**